ibis/expr/tests/test_rules.py

```diff
@@ -46,10 +46,10 @@ def test_invalid_datatype(value, expected):
         assert rlz.datatype(value)
 
 
-def test_invalid_literal():
-    msg = "Value 1 cannot be safely coerced to `string`"
-    with pytest.raises(TypeError, match=msg):
-        rlz.literal(dt.string, 1)
+def test_string_literal_from_intergerinvalid_literal():
+    lit = rlz.literal(dt.string, 1)
```

Comment:
Looks like we're changing the behavior to cast the integer to a string?

Reply:
Yes, ibis.literal(1, dt.string) is constructing an ops.Literal('1', dt.string) now

Reply:
Since it is "coercible"